osu.Game.Rulesets.Osu/Difficulty/Skills/OsuStrainSkill.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using osu.Game.Rulesets.Difficulty.Preprocessing;
+using osu.Game.Rulesets.Difficulty.Skills;
+using osu.Game.Rulesets.Mods;
+using System.Linq;
+
+namespace osu.Game.Rulesets.Osu.Difficulty.Skills
+{
+    public abstract class OsuStrainSkill : StrainSkill
+    {
+        protected virtual int ReducedSectionCount => 10;
+        protected virtual double ReducedStrainBaseline => 0.7;
+        protected virtual double DifficultyMultiplier => 1.08;
+
+        public OsuStrainSkill(Mod[] mods) : base(mods)
+        {
+        }
+
+        public double OsuDifficultyValue()
+        {
+            double difficulty = 0;
+            double weight = 1;
+
+            double strainMultiplier;
+            List<double> strains = GetCurrentStrainPeaks().OrderByDescending(d => d).ToList();
+
+            for (int i = 0; i < ReducedSectionCount; i++)
+            {
+                strainMultiplier = ReducedStrainBaseline + Math.Log10(i * 9.0 / ReducedSectionCount + 1) * (1.0 - ReducedStrainBaseline);
```

Comment:
Is the `9.0` here somehow related to the `ReducedSectionCount`? If so, it should be replaced by `(ReducedSectionCount - 1)`. Possibly with a `float` cast somewhere to avoid loss of precision.

Reply:
It's related to the log function. I want it such that when `i` is equal to `ReducedSectionCount` we always get a value of `1.0` with any `ReducedSectionCount` parameter.
It was `Math.Log10(i * 9.0 + 1)` essentially to make it `f(1.0) = 1.0` and then I divided by `ReducedSectionCount` to scale it on the x axis.

Reply:
Right, I see - the 10 is from the base of the log. I'd probably add an explanatory comment that `strainMultiplier(ReducedSectionCount) = 1` was indeed the intention here.

Reply:
Or, you could alternatively even calculate that log argument with
```csharp
Interpolation.Lerp(1, 10, Math.Clamp((float)i / ReducedSectionCount, 0, 1));
```
if I'm not mistaken.

Reply:
> Or, you could alternatively even calculate that log argument with
> 
> ```cs
> Interpolation.Lerp(1, 10, Math.Clamp((float)i / ReducedSectionCount, 0, 1));
> ```
> 
> if I'm not mistaken.

That seems correct, I'll take it.